src/Utils/ShippingDateFilter.php

```diff
@@ -52,11 +52,18 @@ public function accept(OrderInterface $order, TsRange $range, \DateTime $now = n
 
         $timeline = $this->orderTimelineCalculator->calculate($order, $range);
         $preparation = $timeline->getPreparationExpectedAt();
-
-        if ($preparation <= $now) {
-
-            $this->logger->info(sprintf('ShippingDateFilter::accept | preparation time "%s" is in the past',
-                $preparation->format(\DateTime::ATOM)),
+        // if a customer wants a dropoff at 12:00, and preparation time + shipping time is 30 minutes, I need to order at 11:30 at least
+        // if the dispatcher adds an delay of 30min
+        // then I want here to filter out all of the possible dropoffs between 11:30 and 12:00
+        // so I need to substract the delay from the preparation time
+        $preparationWithDelay = clone $preparation;
+        $preparationWithDelay = $preparationWithDelay->sub(date_interval_create_from_date_string(sprintf('%s minutes', $orderingDelayMinutes)));
+
+        if ($preparationWithDelay <= $now) {
```

Comment:
It's exactly the same logic, but maybe it's a bit easier to understand:

```
$preparationCanStartAt = clone $now;
if ($orderingDelayMinutes > 0) {
    $preparationCanStartAt = $preparationCanStartAt->add(date_interval_create_from_date_string(sprintf('%s minutes', $orderingDelayMinutes)));
}
if ($preparation <= $preparationCanStartAt) {
```

Reply:
cool thx it is nice

in this case which variable do we need to pass on l.82 according to you? (where I added a comment "I am not sure") I think it is still "preparationWithDelay" no?

Reply:
I'd say `$preparation`. We'll filter out all options before `$preparationCanStartAt` on line 62, and on line 82 we only need to check that the restaurant is still opened at the expected `$preparation` time

Reply:
i gave it a second thought, it is `$preparationWithDelay` because delay is the delay bar but also "prior notice" setting on the restaurant page, so you want the restaurant to be open "some time in advance before the preparation can start"